scrapy/core/http2/stream.py

```diff
@@ -431,7 +431,8 @@ def close(
             errors.insert(0, InactiveStreamClosed(self._request))
             self._deferred_response.errback(ResponseFailed(errors))
 
-        elif reason is StreamCloseReason.INVALID_HOSTNAME:
+        else:
+            assert reason is StreamCloseReason.INVALID_HOSTNAME
```

Comment:
What about #4440? :smile:

Reply:
That was triggered by [this comment by me](https://github.com/scrapy/scrapy/pull/4054#discussion_r351514071) (emphasis added):

> We should not use assert statements in production code, outside tests, **where we know the assertion may fail**.

In this case, no input could ever make that `assert` raise an `AssertionError`. The code is written so that this condition will always be `True`. If that ever changes, the cause will be a bug in the code, not unexpected input.

*But*, if you prefer, we could just replace it with a comment besides the `else`. However, if we ever extend the possible values of `StreamCloseReason` and forget to update this part of the code accordingly, an `assert` could cause tests to fail.

Reply:
Fair enough, thanks for the explanation :slightly_smiling_face: